### PR TITLE
Optimized Component Ejector

### PR DESCRIPTION
**Pull request by pupnewfster**
## Changes proposed in this pull request:
- Fixes Digital miner not combining stacks for ejecting
- Combine TransitRequests for ComponentEjector where possible, and only eject once a second instead of also every time an operation occurs.
- Removed ComponentEjector trackers as from what I can tell they did not actually do anything

Some profiling hints from my test world:
Before: https://sparkprofiler.github.io/#hksdBcW8EO https://gyazo.com/32ac3e0956481b904fa2accc194d1990
After: https://sparkprofiler.github.io/#pUOsQgm7zP https://gyazo.com/497480ae84fdd6f1750139009877c230

### CODE REVIEW

File: src/main/java/mekanism/common/tile/component/TileComponentEjector.java [hunk 1]
```diff
@@ -139,52 +123,53 @@ private void ejectFluid(Set<EnumFacing> outputSides, FluidTank tank) {
...
+                if (ejectMap.isEmpty()) {
                     break;
                 }
             }
+            BlockPos offsetPos = tileEntity.getPos().offset(side);
+            TileEntity tile = tileEntity.getWorld().isBlockLoaded(offsetPos) ? tileEntity.getWorld().getTileEntity(offsetPos) : null;
+            ILogisticalTransporter capability = CapabilityUtils.getCapability(tile, Capabilities.LOGISTICAL_TRANSPORTER_CAPABILITY, side.getOpposite());
+            TransitResponse response;
+            if (capability == null) {
+                response = InventoryUtils.putStackInInventory(tile, ejectMap, side, false);
```
Comment (thiakil) [inline src/main/java/mekanism/common/tile/component/TileComponentEjector.java]: does this need to be moved outside the side loop?

Comment (thiakil) [reply src/main/java/mekanism/common/tile/component/TileComponentEjector.java]: (might be TransporterUtils.insert that this applies to)

File: src/main/java/mekanism/common/tile/component/TileComponentEjector.java [hunk 2]
```diff
@@ -139,52 +123,53 @@ private void ejectFluid(Set<EnumFacing> outputSides, FluidTank tank) {
...
+        TransitRequest ejectMap = null;
+        for (EnumFacing side : outputs) {
+            if (ejectMap == null) {
+                ejectMap = getEjectItemMap(data);
+                if (ejectMap.isEmpty()) {
                     break;
                 }
             }
+            BlockPos offsetPos = tileEntity.getPos().offset(side);
+            TileEntity tile = tileEntity.getWorld().isBlockLoaded(offsetPos) ? tileEntity.getWorld().getTileEntity(offsetPos) : null;
```
Comment (pupnewfster) [inline src/main/java/mekanism/common/tile/component/TileComponentEjector.java]: If 1.12 is merged back into this, then we can do `tile = MekanismUtils.getTile` to check for the isBlockLoaded in an easier way to read

File: src/main/java/mekanism/common/tile/component/TileComponentEjector.java (hunk 1 above)
Comment (pupnewfster) [reply src/main/java/mekanism/common/tile/component/TileComponentEjector.java]: I believe this is the correct spot as it is trying on specific sides. `InventoryUtils#putStackInInventory` uses tile being inserted to as a param, and `TransporterUtils#insert` uses tile from. The response needs to be in the loop as it the methods require side information.

**Comment by thiakil** [comment]
Merged manually.

I halved the tickDelay for the eject component (half a second from a full second) - ejecting is now a lot slower since it's only timer based. We could consider doing the ejection at the end of a tile's `tick` method instead if there are complaints

**Pull request closed without merging**